packages/configuration-screen/src/setupConfiguration.ts

```diff
@@ -0,0 +1,141 @@
+    var configurationFileName = 'app';
+    var stanzaName = 'install';
+    var propertiesToUpdate = {
+        is_configured: 'true',
+    };
+
+    await updateConfigurationFile(
+        splunkService,
+        configurationFileName,
+        stanzaName,
+        propertiesToUpdate
+    );
+}
+
+async function reloadApp(splunkService: SplunkService, appName: string): Promise<void> {
+    var splunkApps = splunkService.apps();
+    await promisify(splunkApps.fetch)();
+
+    var currentApp = splunkApps.item(appName);
+    await promisify(currentApp.reload)();
+}
+
+function redirectToHomepage(appName: string): void {
+    var redirectUrl = '/app/' + appName;
+    window.location.href = redirectUrl;
+}
+
+function createService(applicationNamespace: SplunkApplicationNamespace): SplunkService {
+    var http = new splunkjs.SplunkWebHttp();
+    var service = new splunkjs.Service(http, applicationNamespace);
+
+    return service;
+}
+
+function retrieveUserTenants(
+    serverKey: string,
+    successCallback: (userTenants: Array<Tenant>) => void,
+    errorCallback: (errorMessage: string) => void
+): void {
+    const service = createService(applicationNameSpace);
+    const data = { serverKey: serverKey };
+    service.post('/services/retrieve_user_tenants', data, function (err, response) {
+        if (err) {
+            errorCallback(err.data);
+        } else if (response.status === 200) {
+            successCallback(response.data.tenants);
+        }
+    });
+}
+
+async function saveConfiguration(serverKey: string, tenantId: string) {
+    try {
+        const service = createService(applicationNameSpace);
+
+        const storagePasswords = await promisify(service.storagePasswords().fetch)();
+        await savePassword(storagePasswords, storageRealm, 'serverkey', serverKey);
+        await savePassword(storagePasswords, storageRealm, 'tenantid', tenantId);
+
+        await completeSetup(service);
+        await reloadApp(service, appName);
+
+        redirectToHomepage(appName);
+    } catch (error) {
+        console.log('Error:', error);
+    }
+}
+
+async function retrieveServerKey() {
+    return retrievePassword('serverkey');
+}
+
+async function retrieveTenantId() {
+    return retrievePassword('tenantid');
+}
+
+async function retrievePassword(passwordKey: string) {
+    const service = createService(applicationNameSpace);
+    const storagePasswords = await promisify(service.storagePasswords().fetch)();
+    var passwordList = storagePasswords.list();
+    const passwordId = storageRealm + ':' + passwordKey + ':';
+
+    for (var index = 0; index < passwordList.length; index++) {
+        if (passwordList[index].name === passwordId) {
+            return passwordList[index]._properties.clear_password;
+        }
+    }
+    return '';
+}
+
+function doesPasswordExist(
+    storage: SplunkStoragePasswordAccessors,
+    storageRealm: string,
+    key: string
+) {
+    const passwordId = storageRealm + ':' + key + ':';
```

Comment:
String concatenation is so old school. Are you a ye or a ne on template literals? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

Reply:
A big ye, will change it!